internal/crypto/src/time_stamp/verify.rs

```diff
@@ -0,0 +1,369 @@
+// Copyright 2022 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License,
+// Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+// or the MIT license (http://opensource.org/licenses/MIT),
+// at your option.
+
+// Unless required by applicable law or agreed to in writing,
+// this software is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or
+// implied. See the LICENSE-MIT and LICENSE-APACHE files for the
+// specific language governing permissions and limitations under
+// each license.
+
+use std::ops::Deref;
+
+use async_generic::async_generic;
+use bcder::{decode::SliceSource, encode::Values, ConstOid, OctetString};
+use chrono::{offset::LocalResult, DateTime, TimeZone, Utc};
+use x509_certificate::{
+    asn1time::{GeneralizedTime, GeneralizedTimeAllowedTimezone, Time},
+    DigestAlgorithm,
+};
+
+use crate::{
+    asn1::{
+        rfc3161::TstInfo,
+        rfc5652::{
+            CertificateChoices::Certificate, SignerIdentifier, OID_MESSAGE_DIGEST, OID_SIGNING_TIME,
+        },
+    },
+    raw_signature::validator_for_sig_and_hash_algs,
+    time_stamp::{
+        response::{signed_data_from_time_stamp_response, tst_info_from_signed_data},
+        TimeStampError,
+    },
+};
+
+/// Decode the TimeStampToken info and verify it against the supplied data.
+///
+/// TEMPORARILY PUBLIC while refactoring
+#[async_generic]
+pub fn verify_time_stamp(ts: &[u8], data: &[u8]) -> Result<TstInfo, TimeStampError> {
+    // Did the time stamp expire between issuance and verification?
+    let Some(sd) = signed_data_from_time_stamp_response(ts)? else {
```

Comment:
this sigTst2 compatibility this should be:  

 let Ok(Some(sd)) = get_timestamp_signed_data(ts) else {
        return Err(Error::CoseInvalidTimeStamp);
    };

Reply:
I updated `signed_data_from_timestamp_response` to include the changes you had made in `get_timestamp_signed_data`.